### ISSUE TITLE
[BUG] Trace configured to 'on' is causing test failures on Gitlab CI

### ISSUE THREAD

**Issue by mahtabnejad90** [microsoft/playwright#26604]
### System info
- Playwright Version: v1.37.1
- Operating System: saas-linux-medium-amd64
- Browser: Chromium
- Other info: Gitlab CI

### Source code

- N/A

**Link to the GitHub repository with the repro**

N/A

or

**Config file**

```js
import { PlaywrightTestConfig, devices } from '@playwright/test';

// eslint-disable-next-line @typescript-eslint/no-var-requires
require('dotenv').config({ path: 'local.env' });
/**
 * Read environment variables from file.
 * https://github.com/motdotla/dotenv
 */
// require('dotenv').config();

/**
 * See https://playwright.dev/docs/test-configuration.
 */
const config: PlaywrightTestConfig = {
  testDir: './src/tests',
  /* Maximum time one test can run for. */
  timeout: 120 * 1000,
  globalTimeout: 1000 * 1000,
  expect: {
    /**
     * Maximum time expect() should wait for the condition to be met.
     * For example in `await expect(locator).toHaveText();`
     */
    timeout: 100000,
  },
  /* Run tests in files in parallel */
  fullyParallel: true,
  /* Fail the build on CI if you accidentally left test.only in the source code. */
  forbidOnly: !!process.env.CI,
  /* Retry on CI only */
  retries: process.env.CI ? 3 : 3,
  /* Opt out of parallel tests on CI. */
  workers: process.env.CI ? 2 : undefined,
  /* Reporter to use. See https://playwright.dev/docs/test-reporters */
  reporter: process.env.CI
    ? [['list'], ['html'], ['@currents/playwright']]
    : [['list'], ['html']],
  /* Shared settings for all the projects below. See https://playwright.dev/docs/api/class-testoptions. */
  use: {
    headless: true,
    viewport: { width: 1280, height: 720 },
    actionTimeout: 100000,
    ignoreHTTPSErrors: true,
    video: 'on',
    screenshot: 'on',
    /* Collect trace when retrying the failed test. See https://playwright.dev/docs/trace-viewer */
    trace: 'on',
  },

  /* Configure projects for major browsers */
  projects: [
    {
      name: 'chromium',
      use: {
        ...devices['Desktop Chrome'],
      },
    },

    {
      name: 'firefox',
      use: {
        ...devices['Desktop Firefox'],
      },
    },

    {
      name: 'webkit',
      use: {
        ...devices['Desktop Safari'],
      },
    },
    {
      name: 'Edge',
      use: {
        channel: 'msedge',
      },
    },
    {
      name: 'Chrome',
      use: {
        channel: 'chrome',
      },
    },

    /* Test against mobile viewports. */
    // {
    //   name: 'Mobile Chrome',
    //   use: {
    //     ...devices['Pixel 5'],
    //   },
    // },
    // {
    //   name: 'Mobile Safari',
    //   use: {
    //     ...devices['iPhone 12'],
    //   },
    // },

    /* Test against branded browsers. */
    //moved edge before safari
  ],

  /* Folder for test artifacts such as screenshots, videos, traces, etc. */
  outputDir: 'test-results/',

  /* Run your local dev server before starting the tests */
  // webServer: {
  //   command: 'npm run start',
  //   port: 3000,
  // },
};

export default config;
```

**Test file (self-contained)**

```js
test.describe.serial('Offices Submission Flows Happy Paths', () => {
  test.beforeAll(async ({ browser }) => {
    page = await browser.newPage();
    sidebar = new FirstbaseAppAdminSidebar(page);
    firstbaseLoginPage = new FirstbaseAppLoginPage(page);
    await page.goto(process.env.PLAYWRIGHT_MANAGEMENT_BASEURL as string);
    mgmtToken = await GraphQLHelpers.mgmtToken(
      page,
      process.env.PLAYWRIGHT_ADMIN_USERNAME_SIX as string,
      process.env.PLAYWRIGHT_ADMIN_PASSWORD as string,
    );
    organization = await new Organization(
      'add-submit-new-office-happy',
      mgmtToken,
    );
    await organization.create();

    admin = await new Person(
      true,
      process.env.PLAYWRIGHT_COUNTRY_CODE as string,
      organization.orgData.slug,
      mgmtToken,
      page,
    );
    await admin.create();
    await admin.activate(await admin.personData.email);
    sidebar = new FirstbaseAppAdminSidebar(page);
    offices = new OfficesPage(page);
    adminAddressPage = new AdminAddressPage(page);
    sharedAdminFunctions = new SharedAdminFunctions(page);
    firstbaseLoginPage = new FirstbaseAppLoginPage(page);
    await sidebar.peopleAndOfficesDropdownButton.click()
    await sidebar.offices.click()

  });

  test.skip(
    "QA-1181 - Adding new US office in office's page (happy path)" +
      Tags.regression(Tags.offices, {
        disableLocalHybrid: true,
      }),
    async () => {
      await firstbaseLoginPage.firstbaseLogin(
        await admin.personData.email,
        process.env.PLAYWRIGHT_BASIC_PASSWORD as string,
      );
      await page.waitForLoadState();
      await sidebar.peopleAndOfficesDropdownButton.click();
      await sidebar.offices.click();
      await page.waitForLoadState();
      test.setTimeout(200000);
      const officeName =
        'QA-1181' + faker.word.adverb() + ' US ' + faker.word.noun();
      await offices.addOfficeButton.click();
      await offices.officeNameTextField.click();
      await offices.officeNameTextField.fill(officeName);
      await adminAddressPage.addressCountrySelectBox.click();
      await sharedAdminFunctions.modalBodyElement
        .getByText('United States', { exact: true })
        .click();
      await page.waitForLoadState();
      await adminAddressPage.addressStateSelectBox.click();
      await adminAddressPage.stateNevadaSelectOption.click();
      await adminAddressPage.addressLineOneTextField.click();
      await adminAddressPage.addressLineOneTextField.fill(
        'Test Automation Lane 1',
      );
      await adminAddressPage.addressCityNameTextField.click();
      await adminAddressPage.addressCityNameTextField.fill('Las Vegas');
      await adminAddressPage.zipCodeFieldInput.click();
      await adminAddressPage.zipCodeFieldInput.fill('12345');
      await adminAddressPage.addressPhoneNumberTextField.click();
      await adminAddressPage.addressPhoneNumberTextField.fill(
        '1111-2222-3333-4444',
      );
      await sharedAdminFunctions.modalSubmitButton.click();
      await expect(
        offices.officeSubmissionSuccessBanner(officeName),
      ).toBeVisible();
      await page.waitForLoadState();
      await expect(page).toHaveURL(mainEndpoints.officesEndpoint);
    },
  );

  test(
    "QA-1311 - Adding new UK office in office's page (happy path)" +
      Tags.regression(Tags.offices, {
        disableLocalHybrid: true,
        disableDev: true,
        disableStaging: true
      }),
    async () => {
      test.setTimeout(200000);
      const officeName =
        'QA-1311' + faker.word.adverb() + ' UK ' + faker.word.noun();
      await offices.addOfficeButton.click();
      await offices.officeNameTextField.click();
      await offices.officeNameTextField.fill(officeName);
      await adminAddressPage.addressCountrySelectBox.click();
      await sharedAdminFunctions.modalBodyElement
        .getByText('United Kingdom', { exact: true })
        .click();
      await page.waitForLoadState();
      await adminAddressPage.addressCountyTextField.click();
      await adminAddressPage.addressCountyTextField.fill('Greater London');
      await adminAddressPage.addressLineOneTextField.click();
      await adminAddressPage.addressLineOneTextField.fill(
        'Test Automation Lane 1',
      );
      await adminAddressPage.addressCityNameTextField.click();
      await adminAddressPage.addressCityNameTextField.fill('London');
      await adminAddressPage.postCodeFieldInput.click();
      await adminAddressPage.postCodeFieldInput.fill('SW8 9AW');
      await adminAddressPage.addressPhoneNumberTextField.click();
      await adminAddressPage.addressPhoneNumberTextField.fill(
        '1111-2222-3333-4444',
      );
      await sharedAdminFunctions.modalSubmitButton.click();
      await expect(
        offices.officeSubmissionSuccessBanner(officeName),
      ).toBeVisible();
      await page.waitForLoadState();
      await expect(page).toHaveURL(mainEndpoints.officesEndpoint);
    },
  );

  test(
    "QA-1312 - Adding new Canada office in office's page (happy path)" +
      Tags.regression(Tags.offices, {
        disableLocalHybrid: true,
      }),
    async () => {
      test.setTimeout(200000);
      const officeName =
        'QA-1312' + faker.word.adverb() + ' CA ' + faker.word.noun();
      await offices.addOfficeButton.click();
      await offices.officeNameTextField.click();
      await offices.officeNameTextField.fill(officeName);
      await adminAddressPage.addressCountrySelectBox.click();
      await sharedAdminFunctions.modalBodyElement
        .getByText('Canada', { exact: true })
        .click();
      await page.waitForLoadState();
      await adminAddressPage.addressStateSelectBox.click();
      await page.getByText('British Columbia', { exact: true }).click();
      await adminAddressPage.addressLineOneTextField.click();
      await adminAddressPage.addressLineOneTextField.fill(
        'Test Automation Lane 1',
      );
      await adminAddressPage.addressCityNameTextField.click();
      await adminAddressPage.addressCityNameTextField.fill('London');
      await adminAddressPage.postCodeFieldInput.click();
      await adminAddressPage.postCodeFieldInput.fill('M4B 1G5');
      await adminAddressPage.addressPhoneNumberTextField.click();
      await adminAddressPage.addressPhoneNumberTextField.fill(
        '1111-2222-3333-4444',
      );
      await sharedAdminFunctions.modalSubmitButton.click();
      await expect(
        offices.officeSubmissionSuccessBanner(officeName),
      ).toBeVisible();
      await page.waitForLoadState();
      await expect(page).toHaveURL(mainEndpoints.officesEndpoint);
    },
  );

  test(
    "QA-1313 - Adding new other country (i.e Sweden) office in office's page (happy path)" +
      Tags.regression(Tags.offices, {
        disableLocalHybrid: true,
      }),
    async () => {
      test.setTimeout(200000);
      const officeName =
        'QA-1313' + faker.word.adverb() + ' SE ' + faker.word.noun();
      await offices.addOfficeButton.click();
      await offices.officeNameTextField.click();
      await offices.officeNameTextField.fill(officeName);
      await adminAddressPage.addressCountrySelectBox.click();
      await sharedAdminFunctions.modalBodyElement
        .getByText('Sweden', { exact: true })
        .click();
      await page.waitForLoadState();
      await adminAddressPage.addressRegionTextField.click();
      await adminAddressPage.addressRegionTextField.fill('Scania');
      await adminAddressPage.addressLineOneTextField.click();
      await adminAddressPage.addressLineOneTextField.fill(
        'Test Automation Lane 1',
      );
      await adminAddressPage.addressCityNameTextField.click();
      await adminAddressPage.addressCityNameTextField.fill('Stockholm');
      await adminAddressPage.postCodeFieldInput.click();
      await adminAddressPage.postCodeFieldInput.fill('17000');
      await adminAddressPage.addressPhoneNumberTextField.click();
      await adminAddressPage.addressPhoneNumberTextField.fill(
        '1111-2222-3333-4444',
      );
      await sharedAdminFunctions.modalSubmitButton.click();
      await expect(
        offices.officeSubmissionSuccessBanner(officeName),
      ).toBeVisible();
      await page.waitForLoadState();
      await expect(page).toHaveURL(mainEndpoints.officesEndpoint);
    },
  );
});
```

**Steps**
- Ensure `trace: 'on'` in `playwright.config.ts`
- Run tests in Gitlab CI

**Expected**

Test should pass on Gitlab CI

**Actual**

Test fails and henceforth CI fails after timeout period and we also see timeout errors below job logs

![Screenshot 2023-08-22 at 15 02 58](https://github.com/microsoft/playwright/assets/41478863/fd01f903-d505-4173-87cf-3e70f686ab38)

![Screenshot 2023-08-22 at 15 28 18](https://github.com/microsoft/playwright/assets/41478863/bb0c4578-3423-45fb-8797-c09bb1879b7d)

This cannot be reproduced on local machine.

**Comment by yury-s** [comment]
Since the error does not reproduce locally, it must have something to do with the gitlab environment. Since the test times out, it might be that you need to increase the global timeout on CI (e.g. if the tests take 800s to finish locally it may well be the case that you need more time on CI). Another thing is the network configuration could be different and some of the requests that go through locally may be failing there. Since this is a gitlab problem, maybe you can prepare a self contained repro in a docker? Without a repro that we could run locally we cannot do anything about it on the playwright end.

Please file a new issue with a repro and link it to this one if you think there is a bug in playwright.
